src/census_response.py

```diff
@@ -380,42 +383,24 @@ def il_county_filter(x):
     response_json = response.json()
 
     if reverse:
-        il_json = {county[1]+county[2]: county[0]
+        il_json = {county[1] + county[2]: county[0]
                    for county in response_json if il_county_filter(county)}
     else:
-        il_json = {county[0]: county[1]+county[2]
+        il_json = {county[0]: county[1] + county[2]
                    for county in response_json if il_county_filter(county)}
 
     return il_json
 
 
-def bin_data(geo_data: dict):
-    '''
-    Creates bins from functions described in this function
-
-    geoData:
-        keys: geo code
-        values: percentages or data to bin
-    '''
-
-    def quartiles(geo_value):
-        min_v = np.min(geo_value)
-        first = np.quantile(geo_value, 0.25)
-        second = np.quantile(geo_value, 0.5)
-        third = np.quantile(geo_value, 0.75)
-        max = np.max(geo_value)
-        q = [min_v, first, second, third, max]
-
-        return {'quartiles': q}
-
+def calculate_natural_breaks_bins(df: pd.DataFrame, bin_count: int, column_names: List[str]) -> Dict[str, List[float]]:
+    """
+    :param df: Pandas dataframe.
+    :param bin_count: Number of bins used to classify data.
+    :param column_names The dataframe column names to use to calculate jenks natural breaks
+    :return: Dictionary of column name and list of bin cutoff limits.
+    """
     bin_dict = {}
-
-    for k in geo_data:
-        v = geo_data[k]
-        try:
-            q = quartiles(v)
-        except ValueError:
-            continue
-        bin_dict[k] = q
-
+    for cn in column_names:
+        column_data = df[cn].dropna().to_list()
```

Comment:
There are some NaN sneaking into the input. I removed them here, but they should be cleansed upstream.

Reply:
The NaNs are from zip codes with no population (think it is just one). Do you think it makes more sense to drop that zip code rather to have null for the percentages?

Reply:
I would drop the zip code. I think 0 populations could be one of two things: 1. Actual 0 population. i.e. an industrial area 2. Data entry error. 

I think it would make sense to remove the 0 population zip codes because they feel like outliers that could throw other calculations off. Alternatively, there could be a flag (maybe ENV VAR) that allows the user to toggle between functions using 0 population or not.

For this calculation, removing 0 population zip codes would be helpful. But I do not know if they are needed elsewhere.